x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/ApiKeyService.java

```diff
@@ -299,22 +301,44 @@ public void createApiKey(
         if (authentication == null) {
             listener.onFailure(new IllegalArgumentException("authentication must be provided"));
         } else {
-            createApiKeyAndIndexIt(authentication, request, userRoleDescriptors, listener);
+            final TransportVersion version = getMinNodeTransportVersion();
+            if (version.before(RoleDescriptor.VERSION_REMOTE_INDICES) && hasRemoteIndices(request.getRoleDescriptors())) {
```

Comment:
@n1v0lg I was working on some integration tests which made me realize that this check is probably wrong?
The `RoleDescriptor.VERSION_REMOTE_INDICES` points to 8.6.0 version, but this change is going to land in 8.8.0.
Are we planning to update `RoleDescriptor.VERSION_REMOTE_INDICES` before we go GA or will this constant stay fixed?

Reply:
I adjusted the version check and  set it to 8.8.0 (https://github.com/elastic/elasticsearch/pull/93414/commits/cdf6d0a66e9f2256b8ac5bee5fc1b2e245d81e7c).